src/vs/base/common/glob.ts

```diff
@@ -264,6 +268,16 @@ export function match(arg1: string | IExpression, path: string, siblings?: strin
 
 	// Glob with String
 	if (typeof arg1 === 'string') {
+
+		if (trivia1.test(arg1)) {
+			// common pattern: **/*.txt just need endsWith check
+			return strings.endsWith(path, arg1.substr(4)); // '**/*'.length === 4
```

Comment:
Plus current implementation has an issue, for example if I will have two file types `foo.myext` and `bar.notmyext` and have a pattern `**/*.myext` - both files will match this extension.

Please add unit tests for this improvement.